File: backend/app/schemas/practice.py
import uuid
from datetime import datetime
from typing import Any, List, Optional

from pydantic import BaseModel, field_validator


class Team(BaseModel):
    id: str
    title: str
    description: str
    members: List[str] = []


class TeamsConfig(BaseModel):
    teams: List[Team] = []


class TeamCreate(BaseModel):
    title: str
    description: str = ""


class TeamUpdate(BaseModel):
    title: str
    description: str = ""


class TeamMembersUpdate(BaseModel):
    members: List[str]


class PriorityConfig(BaseModel):
    low: str = ""
    medium: str = ""
    high: str = ""


class PracticeUpdate(BaseModel):
    practice_name: Optional[str] = None
    practice_region: Optional[str] = None
    priority_config: Optional[PriorityConfig] = None


class PracticeResponse(BaseModel):
    id: uuid.UUID
    practice_name: str
    practice_region: str
    active_call_ids: List[uuid.UUID]
    max_concurrent_calls: int
    teams: TeamsConfig
    priority_config: PriorityConfig
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

    @field_validator("teams", mode="before")
    @classmethod
    def parse_teams(cls, v: Any) -> TeamsConfig:
        if isinstance(v, dict):
            return TeamsConfig(**v)
        return v

    @field_validator("priority_config", mode="before")
    @classmethod
    def parse_priority_config(cls, v: Any) -> PriorityConfig:
        if isinstance(v, dict):
            return PriorityConfig(**v)
        return v if v else PriorityConfig()

File: frontend/types/practice.ts
export interface Team {
  id: string;
  title: string;
  description: string;
  members: string[];
}

export interface TeamsConfig {
  teams: Team[];
}

export interface TeamCreate {
  title: string;
  description: string;
}

export interface TeamUpdate {
  title: string;
  description: string;
}

export interface TeamMembersUpdate {
  members: string[];
}

export interface PriorityConfig {
  low: string;
  medium: string;
  high: string;
}

export interface Practice {
  id: string;
  practice_name: string;
  practice_region: string;
  active_call_ids: string[];
  max_concurrent_calls: number;
  teams: TeamsConfig;
  priority_config: PriorityConfig;
  created_at: string;
  updated_at: string;
}

export interface PracticeUpdate {
  practice_name?: string;
  practice_region?: string;
  priority_config?: PriorityConfig;
}
